app/jobs/kit_import_job.rb

```diff
@@ -0,0 +1,125 @@
+class KitImportJob < ApplicationJob
+  queue_as :dradis_upload
+
+  rescue_from(StandardError) do |e|
+    logger.info "An error ocurred: #{e.message}"
+  end
+
+  def perform(file:, logger:, user_id: nil)
+    @current_user = user_id ? User.find(user_id) : User.first
+    @file = file
+    @logger = logger
+    @report_templates_dir = Configuration.paths_templates_reports
+    @temporary_dir = Dir.mktmpdir
+
+    FileUtils.cp file, temporary_dir
+    unzip
+    import_methodology_templates
+    import_note_templates
+    import_project_package
+    import_project_templates
+    import_report_template_files
+    if defined?(Dradis::Pro)
+      import_report_template_properties
+      import_rules
+    end
+
+  ensure
+    logger.info('Worker process completed.')
+    FileUtils.remove_entry temporary_dir
+  end
+
+  private
+  attr_reader :current_user, :file, :logger, :report_templates_dir, :temporary_dir
+
+  def import_methodology_templates
+    logger.info { 'Copying methodology templates...' }
+    import_templates('methodologies')
+  end
+
+  def import_note_templates
+    logger.info { 'Copying issue, evidence templates...' }
+    import_templates('notes')
+  end
+
+  def import_project_package
+    logger.info { 'Importing project package...' }
+
+    project_package = Dir.glob("#{temporary_dir}/kit/*.zip").first
+
+    unless project_package
+      logger.info { '  - Project package not found...' }
+      return
+    end
+
+
+    if defined?(Dradis::Pro)
+      project = Project.create(name: File.basename(project_package, '.zip'))
+      if project.errors.any?
+        logger.info { '  - Project errors: '}
+        project.errors.full_messages.each do |error|
+          logger.info { "    - #{error}"}
+        end
+        return
+      end
+      project.assign_owner(current_user)
+    else
+      project = Project.new
+    end
+
+    logger.info { "  - Importing project: #{project.name}" }
+    importer = Dradis::Plugins::Projects::Upload::Package::Importer.new(
+      project_id: project.id,
+      plugin: Dradis::Plugins::Projects::Upload::Package,
+      default_user_id: current_user.id
+    )
+    importer.import(file: project_package)
+
+    logger.info { "  - New Project #{project.id} created." }
+  end
+
+  def import_project_templates
+    logger.info { 'Copying project templates...' }
+    import_templates('projects')
+  end
+
+  def import_report_template_files
+    logger.info { 'Copying report template files...' }
+
+    FileUtils.mkdir_p report_templates_dir
+    %w{
+      excel
+      html_export
+      word
+    }.each do |plugin|
+      reports_dir = "#{temporary_dir}/kit/templates/reports/#{plugin}"
+      FileUtils.cp_r(
+        "#{reports_dir}/.",
+        "#{report_templates_dir}/#{plugin}/"
+      ) if File.exist?(reports_dir)
+    end
+  end
+
+  def import_report_template_properties;end
```

Comment:
I'm not a fan of removing the functionality in this method for the CE version. This may cause some issues when syncing this to Pro. Can we instead include the functionality here and in line 21 we have something like:
```
import_report_template_files if defined?(Dradis::Pro)
```
This way we'd have the same versions in CE and Pro and merging will be easier.

Reply:
I have a valid point here but I am unsure about this change.
We would be referencing a class that does not exist in the codebase (`ReportTemplateProperties`).
My idea was that keeping the method but making it a no-op would be clear enough at merge time.

Reply:
I just realized are are already doing the defined check for `Dradis::Pro`, so I was thinking that we were planning this approach beforehand.

My main concern here is the consistency since in `dradis-plugins` we do already something similar. In [content_service/base.rb](https://github.com/dradis/dradis-plugins/blob/master/lib/dradis/plugins/content_service/base.rb), we do a defined check for `Dradis::Pro`:
```
    include Boards        if defined?(Dradis::Pro)
    include Categories
    include ContentBlocks if defined?(Dradis::Pro)
    include Evidence
    include Issues
    include Nodes
    include Notes
    include Properties    if defined?(Dradis::Pro)
```
and in those modules we are referencing classes that do not exist in the CE codebase too.

Reply:
Ok, restored these methods